services/housekeeper/housekeeper.go

```diff
@@ -164,13 +225,46 @@ func (hk *Housekeeper) updateKnownValidators() {
 
 	for _, v := range validators {
 		pubkey := types.PubkeyHex(v.Validator.Pubkey)
-		err = hk.redis.SetKnownValidator(pubkey, v.Index)
+		err := hk.redis.SetKnownValidator(pubkey, v.Index)
 		if err != nil {
 			log.WithError(err).WithField("pubkey", pubkey).Fatal("failed to set known validator in Redis")
 		}
 	}
 }
 
+func (hk *Housekeeper) fetchValidators() map[types.PubkeyHex]beaconclient.ValidatorResponseEntry {
+	// return the first successful beacon node response
+	var result map[types.PubkeyHex]beaconclient.ValidatorResponseEntry
+	clients := hk.getBeaconClientsByLastResponse()
+
+	var mu sync.Mutex
```

Comment:
remove?

Reply:
talking more about this in the other comment, the issue is that this is just a local lock. also should be a RWLock, but better to use atomic maybe